### ISSUE TITLE
If loading stl binary file

### ISSUE THREAD

**Issue by Sea-DH1** [jscad/OpenJSCAD.org#972]
stl binary loading error
![image](https://user-images.githubusercontent.com/26751491/147464625-e7d7305c-07c6-4cb4-b1ad-36f258247cf0.png)

[123.txt](https://github.com/jscad/OpenJSCAD.org/files/7779834/123.txt)

This file is a stl file, you need to modify the suffix after downloading

**Comment by z3dev** [comment]
@Sea-DH1 I tried the original STL file at www.openjscad.xyz, and it rendered properly. I also examined the STL file with Meshlab, and didn't find any issues with the STL file. There are problems with the design, which has a lot of 'non-solid' areas.

So, I'm wondering if the file is coming into STL deserializer as 'binary'. I can see that there's a Vue component working here, so I presume that the file is being read from a website, or drag-n-dropped to a webpage.

[123-2.stl.txt](https://github.com/jscad/OpenJSCAD.org/files/7832494/123-2.stl.txt)

The core library has a function that reads the contents of files based on file extension. If the file is considered to contain binary data then the file is read as an ArrayBuffer. See walkFileTree.js here https://github.com/jscad/OpenJSCAD.org/tree/master/packages/core/src/web

Of course, you will have to adjust YOUR code to read the STL file into memory as an ArrayBuffer.
